go-app/main.go

```diff
@@ -172,3 +176,63 @@ func loadAnalyseLog(w http.ResponseWriter, r *http.Request) {
 	}
 	reportTempl.Execute(w, report.FullLogDetails.Analysis_details)
 }
+func loadEventDetails(w http.ResponseWriter, r *http.Request) {
+	r.ParseMultipartForm(10 << 20)
+	startIndex, _ := strconv.Atoi(r.FormValue("StartIndex"))
+	endIndex, _ := strconv.Atoi(r.FormValue("EndIndex"))
+	logs := strings.Split(report.FullLogDetails.Analysis_details.RawLog, "\n")
+	type Reponse struct {
+		Content   []string
+		Highlight map[int]bool //index=> true = must be highlight
+	}
+	event_content := []string{}
+	highlight := map[int]bool{}
+	if startIndex < 0 {
+		startIndex = 0
+	}
+	if endIndex > len(logs) {
+		endIndex = len(logs)
+	}
+	event_content = fillEventDetails(startIndex, endIndex, logs, event_content, highlight)
+	resp := Reponse{
+		Content:   event_content,
+		Highlight: highlight,
+	}
+	jsonValue, _ := json.Marshal(resp)
+	w.Write(jsonValue)
+
+}
+func fillEventDetails(startIndex int, endIndex int, logs []string, event_content []string, highlight map[int]bool) []string {
+	last_index := startIndex
+	for _, index := range report.OrderedEventsLine {
```

Comment:
This is confusing to me, shouldn't the index be the first parameter? It looks like the value you're using for index is the index of the event line in the raw log to me. But that makes the naming confusing since index is usually reserved for the name of the index of what you're iterating through.

Reply:
Okay Got it. I was using index to reference the index of the event line as you said. I did not have in mind that "index" is used for first parameter